web/src/user_topics.js

```diff
@@ -73,9 +100,30 @@ export function set_user_topic(user_topic) {
     switch (user_topic.visibility_policy) {
         case visibility_policy.MUTED:
             add_muted_topic(stream_id, topic, date_muted);
+            update_topic_visibility_policy(
+                stream_id,
+                topic,
+                date_muted,
+                user_topic.visibility_policy,
+            );
```

Comment:
You should be deleting the `add_muted_topic` calls here -- muted topics is a special case of visibility policies, so you're doing the same thing twice in a row with different API endpoints here.